pandas/core/ops.py

```diff
@@ -407,8 +407,12 @@ def _validate_datetime(self, lvalues, rvalues, name):
 
             # if tz's must be equal (same or None)
             if getattr(lvalues, 'tz', None) != getattr(rvalues, 'tz', None):
-                raise ValueError("Incompatible tz's on datetime subtraction "
-                                 "ops")
+                if len(rvalues) == 1 and isna(rvalues).all():
```

Comment:
why does this matter if it is exactly len == 1, seems odd

Reply:
The issue at hand is that `pd.NaT` is special because it can play the role of a `datetime` or a `timedelta` depending on context.  `_TimeOp` wraps scalar inputs in an array that becomes `rvalues`.  But at a basic level here what we are interested in is "was the original `right` argument `pd.NaT`?"  What this is doing (and the same you're asking me to do below) is to undo wrapping done by `_TimeOp`.  It is a _much less clear_ way of checking "is the argument `pd.NaT`?"

Reply:
you didn't answer the question, whether I have 1 NaT or all NaT is immaterial, I cann't use the rhs dtype as its ambiguous and must use the left.